### PR TITLE
[WFLY-5151] and [WFLY-4661] Support mdb delivery groups and mdb clustered singleton

### PR DESCRIPTION
**Pull request by fl4via**

### CODE REVIEW

**Comment by n1hility** [comment]
Retest this please

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/MdbDeliveryGroupRemove.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) {
+        context.reloadRequired();
+    }
+
+    @Override
+    protected void recoverServices(final OperationContext context, final ModelNode operation, final ModelNode model)
+            throws OperationFailedException {
+        try {
+            MdbDeliveryGroupAdd.INSTANCE.installServices(context, operation, model);
```
Comment (stuartwdouglas) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/MdbDeliveryGroupRemove.java]: Shouldn't this just revert the reload required, as it does not actually remove the services.

Comment (stuartwdouglas) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/MdbDeliveryGroupRemove.java]: Also you can probably just use org.jboss.as.controller.ReloadRequiredRemoveStepHandler

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/MdbDeliveryGroupResourceDefinition.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * {@link org.jboss.as.controller.ResourceDefinition} for mdb delivery group.
+ *
+ * @author Flavia Rainone
+ */
+public class MdbDeliveryGroupResourceDefinition extends SimpleResourceDefinition {
+
+    private static final ServiceName DELIVERY_GROUP_SERVICE_NAME = ServiceName.of("org", "wildfly", "ejb3", "mdb", "delivery", "group");
+    public static final SimpleAttributeDefinition ACTIVE = new SimpleAttributeDefinitionBuilder(
+            EJB3SubsystemModel.MDB_DELVIERY_GROUP_ACTIVE, ModelType.BOOLEAN, true).setDefaultValue(new ModelNode(true))
```
Comment (stuartwdouglas) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/MdbDeliveryGroupResourceDefinition.java]: This should allow expressions

**Comment by n1hility** [comment]
Retest this please

**Comment by n1hility** [comment]
retest this please

**Comment by n1hility** [comment]
lol NOW it decides to rerun

**Comment by n1hility** [comment]
Retest this please

**Pull request merged**